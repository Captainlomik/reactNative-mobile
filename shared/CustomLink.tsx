import { Link } from "expo-router";
import { Text, View, StyleSheet } from "react-native";
import { Colors, Fonts } from "./tokens";
import { LinkProps } from "expo-router/build/link/Link";

export function CustomLink({ text, ...props }: LinkProps & { text: string }) {
    return (
        <Link {...props} style={styles.link}>
            <Text>{text}</Text>
        </Link>
    )
}

const styles = StyleSheet.create({
    link: {
        fontSize: Fonts.f18,
        color: Colors.link, 
        fontFamily: Fonts.regular
    }
})