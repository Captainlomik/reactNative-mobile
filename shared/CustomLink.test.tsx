import * as React from "react";
import * as renderer from "react-test-renderer";
import { CustomLink } from "./CustomLink";

describe("<CustomLink />", () => {
  it("simple link", () => {
    const tree = renderer
      .create(<CustomLink href={"/restorej"} text={"Восстановить пароль"} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
