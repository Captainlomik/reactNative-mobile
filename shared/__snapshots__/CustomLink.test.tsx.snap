// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomLink /> smth 1`] = `
<Text
  href="/restorej"
  onPress={[Function]}
  role="link"
  style={
    {
      "color": "#A97BFF",
      "fontFamily": "FiraSans",
      "fontSize": 18,
    }
  }
>
  <Text>
    Восстановить пароль
  </Text>
</Text>
`;
